src/webots/vrml/WbNode.cpp

```diff
@@ -1246,6 +1246,16 @@ void WbNode::exportExternalSubProto(WbWriter &writer) const {
   if (!isProtoInstance())
     return;
 
+  // find all proto that were already exposed prior to converting the root (typically, slots with world visibility)
+  const QList<const WbNode *> protos = WbNodeUtilities::protoNodesInWorldFile(this);
+  foreach (const WbNode *p, protos)
+    // the node itself doesn't need to be re-declared since it won't exist after conversion
+    if (p != this) {
+      const QString protoDeclaration = WbProtoManager::instance()->externProtoUrl(p, false);
+      assert(!protoDeclaration.isEmpty());  // since the proto has world-visibility, a declaration for it must exist
+      writer.trackDeclaration(p->modelName(), protoDeclaration);
+    }
```

Comment:
Here it would be better to add the brackets to the IF block given that it is a multiline content
```suggestion
  foreach (const WbNode *p, protos) {
    // the node itself doesn't need to be re-declared since it won't exist after conversion
    if (p != this) {
      const QString protoDeclaration = WbProtoManager::instance()->externProtoUrl(p, false);
      assert(!protoDeclaration.isEmpty());  // since the proto has world-visibility, a declaration for it must exist
      writer.trackDeclaration(p->modelName(), protoDeclaration);
    }
  }
```